pkg/proxy/endpointslicecache.go

```diff
@@ -290,15 +290,15 @@ func (cache *EndpointSliceCache) addEndpointsByIP(serviceNN types.NamespacedName
 		endpointInfo := newBaseEndpointInfo(endpoint.Addresses[0], portNum, isLocal, endpoint.Topology,
 			endpoint.Ready, endpoint.Serving, endpoint.Terminating, endpoint.ZoneHints)
 
-		// This logic ensures we're deduping potential overlapping endpoints
-		// isLocal should not vary between matching IPs, but if it does, we
+		// This logic ensures we're deduplicating potential overlapping endpoints
+		// isLocal should not vary between matching endpoints, but if it does, we
 		// favor a true value here if it exists.
-		if _, exists := endpointsByIP[endpointInfo.IP()]; !exists || isLocal {
-			endpointsByIP[endpointInfo.IP()] = cache.makeEndpointInfo(endpointInfo)
+		if _, exists := endpointSet[endpointInfo.String()]; !exists || isLocal {
```

Comment:
this questions is mostly curiosity, is `isLocal` really needed now?
we are identifying the endpoint by IP:Port now ...

Reply:
I don't think so either. But this breaks one of the test case. 😟

The only case I come up with that changes the property "isLocal" of the same endpoint is

```
# svc on host1
svc    -> pod1 10.0.0.1:8080 (host1, local)
       -> pod2 10.0.0.2:8080 (host2, non local)

# pods are deleted and rebuilt
svc    -> pod3 10.0.0.1:8080 (host2, non local)
       -> pod4 10.0.0.2:8080 (host1, local)
```

But I don't think it will happen in normal cases because nodes have non-overlapping pod CIDRs ...

Any ideas?

Reply:
yeah, that is weird but we've seen worse :stuck_out_tongue: 
I don' think that it will hurt